lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -101,91 +101,220 @@ export class BrowserStorage extends CacheManager {
     }
 
     /**
-     * Sets the cache item with the key and value given.
-     * Stores in cookie if storeAuthStateInCookie is set to true.
-     * This can cause cookie overflow if used incorrectly.
+     * fetches the entry from the browser storage based off the key
+     * @param key
+     */
+    getItem(key: string): string {
+        return this.windowStorage.getItem(key);
+    }
+
+    /**
+     * sets the entry in the browser storage
      * @param key
      * @param value
      */
-    setItem(key: string, value: string | object, type: string): void {
-        // save the cacheItem
-        switch (type) {
-            case CacheSchemaType.ACCOUNT:
-            case CacheSchemaType.CREDENTIAL:
-            case CacheSchemaType.APP_METADATA:
-            case CacheSchemaType.THROTTLING:
-                this.windowStorage.setItem(key, JSON.stringify(value));
-                break;
-            case CacheSchemaType.TEMPORARY: {
-                const stringVal = value as string;
-                this.windowStorage.setItem(key, stringVal);
-                if (this.cacheConfig.storeAuthStateInCookie) {
-                    this.setItemCookie(key, stringVal);
-                }
-                break;
-            }
-            case CacheSchemaType.TELEMETRY: {
-                this.windowStorage.setItem(key, JSON.stringify(value));
-                break;
-            }
-            default: {
-                throw BrowserAuthError.createInvalidCacheTypeError();
-            }
+    setItem(key: string, value: string): void {
+        this.windowStorage.setItem(key, value);
+    }
+
+    /**
+     * fetch the account entity from the platform cache
+     * @param key
+     */
+    getAccount(key: string): AccountEntity | null {
+        const value = this.getItem(key);
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+
+        const account = CacheManager.toObject(new AccountEntity(), JSON.parse(value));
+        if (AccountEntity.isAccountEntity(account)) {
+            return account;
+        }
+        return null;
+    }
+
+    /**
+     * set account entity in the platform cache
+     * @param key
+     * @param value
+     */
+    setAccount(key: string, value: AccountEntity): void {
+        this.setItem(key, JSON.stringify(value));
+    }
+
+    /**
+     * generates idToken entity from a string
+     * @param value
+     */
+    getIdTokenCredential(key: string): IdTokenEntity {
+        const value = this.getItem(key);
+        if (StringUtils.isEmpty(value)) {
+            return null;
+        }
+        const idToken: IdTokenEntity = CacheManager.toObject(new IdTokenEntity(), JSON.parse(value));
+        return idToken;
+    }
+
+    /**
+     * set IdToken credential to the platform cache
+     * @param key
+     * @param value
+     */
+    setIdTokenCredential(key: string, value: IdTokenEntity): void {
```

Comment:
Do we need a separate setter function for each cache type? They all do the same thing it looks like.

Reply:
For consistency!

Reply:
Agreed to leave it as is for now. If we want to group them in the future, lets revisit it.